### ISSUE TITLE
Structure zmq_event_t was removed

### ISSUE THREAD

**Issue by nkvorn** [aziomq/aziomq4-x#16]
The structure zmq_event_t was removed from ZMQ 4.x from this commit: https://github.com/zeromq/libzmq/commit/9753de8566d335703c96160aa4e5f9c6e55208a9#diff-e81114d231ae71d13f16d1b6970abb37L353

So aziomq4-x cannot be used with current and future ZMQ 4.x...

**Comment by rodgert** [comment]
First, thanks for your interest in aziomq.

It looks like this change affects libzmq4.0.5 and later (I guess that whole not breaking APIs in minor revs thing is a "suggestion" more so than any actual rule), where as I have been building/testing against 4.0.3.

Unfortunately I discovered in testing a couple of deeper flaws in this code and I have been working on putting together a new version of aziomq to address the issues, but haven't been able to complete it due to other time commitments. I hope to finish up the new version and push it out over the next couple of weeks depending on how much time I have next week during the fall C++ Standards Committee meeting. Until then, I would ask that you not use aziomq.

The two specific issues are -
- There is a subtle flaw in how I integrated zeromq sockets into Asio's reactor machinery which means that if you have both async_send and async_receives pending against a socket, and the socket signals, with both events available, one of the two async operations will be ignored. This hadn't showed up in my usage of the library because I tend to do async_receive followed by synchronous send in my designs. The fix for this is somewhat involved.
- I also discovered that I could not completely map zeromq messages to Asio buffers for some important use-cases, so I needed to surface up a message type and extend the socket API to include support for this.

As part of fixing these issues, I restructured the way in which I hang additional functionality like the socket monitor off of a socket, which means I still need to finish the rewrite for that code. I will address this change in libzmq as part of that fix.

Thanks,
Tom.
